test/unit/org/apache/cassandra/cql3/ast/Symbol.java

```diff
@@ -92,12 +95,35 @@ public AbstractType<?> type()
         return type;
     }
 
+    public AbstractType<?> rawType()
+    {
+        if (reversed) return ReversedType.getInstance(type);
+        return type;
+    }
+
     @Override
     public String name()
     {
         return symbol;
     }
 
+    /**
+     * Same as {@link #visit(Visitor)} but has the restriction that the return type may not be changed away from {@link Symbol}.
+     */
+    public Symbol visit(String type, Visitor v)
+    {
+        var u = v.visit(this);
+        if (u == this) return this;
+        if (!(u instanceof Symbol))
+            throw new IllegalStateException("Visitor converted a Symbol to " + u.getClass().getSimpleName() + "; this is not supported with " + type);
+        return (Symbol) u;
+    }
+
+    public String detailedName()
+    {
+        return symbol + " " + type.asCQL3Type() + (reversed ? " (reversed)" : "");
```

Comment:
this has been nice in the test in the next patch... below is a sample of the sai reverse bug

```
		266: SELECT * FROM "arak0ffcoYDbQhwmjr1uXjr".tbl WHERE ck0 <= 0; -- ck0 varint (reversed), indexed with SAI
...
Caused by: java.lang.AssertionError: Missing rows:
pk0                     | pk1                  | ck0        | ck1                                | v0                        
-3.793674428756257E-259 | '11:03:35.709809507' | -153195755 | 0x00000000000013009e00000000000000 | '2027-09-06T19:24:27.334Z'

Expected:
pk0                     | pk1                  | ck0        | ck1                                | v0                        
-3.793674428756257E-259 | '11:03:35.709809507' | 0          | 0x0000000000001000bd00000000000000 | '2053-01-08T04:12:35.057Z'
-3.793674428756257E-259 | '11:03:35.709809507' | -153195755 | 0x00000000000013009e00000000000000 | '2027-09-06T19:24:27.334Z'
```